--- origsrc/yaz-3.0.36/src/Makefile.am	2008-09-03 07:23:57.000000000 -0500
+++ src/yaz-3.0.36/src/Makefile.am	2008-10-02 23:24:48.021463900 -0500
@@ -100,15 +100,15 @@
   iconv_encode_marc8.c iconv_encode_iso_8859_1.c iconv_encode_wchar.c \
   iconv_decode_marc8.c sc.c
 
-libyaz_la_LDFLAGS=-version-info $(YAZ_VERSION_INFO)
+libyaz_la_LDFLAGS=-version-info $(YAZ_VERSION_INFO) -no-undefined
 
 libyaz_server_la_SOURCES = statserv.c seshigh.c eventl.c \
   requestq.c eventl.h session.h mutex.c
 
-libyaz_server_la_LDFLAGS=-version-info $(YAZ_VERSION_INFO)
+libyaz_server_la_LDFLAGS=-version-info $(YAZ_VERSION_INFO) -no-undefined
 
 libyaz_icu_la_SOURCES = icu_I18N.c
-libyaz_icu_la_LDFLAGS=-version-info $(YAZ_VERSION_INFO)
+libyaz_icu_la_LDFLAGS=-version-info $(YAZ_VERSION_INFO) -no-undefined
 
 # Rules for Z39.50 V3
 z-accdes1.c \
